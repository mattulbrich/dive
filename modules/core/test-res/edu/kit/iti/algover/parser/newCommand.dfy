class C {

  method Init(i : int) {}

  method m() {
    var c := new C;
    var c2 : C := new C;
    var c3 := new C.Init(42);

    c := new C;
    c2 := new C.Init(44);
  }
}