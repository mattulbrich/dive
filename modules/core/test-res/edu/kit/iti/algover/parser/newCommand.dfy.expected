(COMPILATION_UNIT
  (class C
    (method Init (ARGS (VAR i (TYPE int))) BLOCK)
    (method m ARGS
      (BLOCK
        (var c (new C))
        (var c2 (TYPE C) (new C))
        (var c3 (new C Init (ARGS 42)))
        (:= c (new C))
        (:= c2 (new C Init (ARGS 44)))))))
